### PR TITLE
[node] support node:[lib] in vc dev

### PR DESCRIPTION
**Pull request by Schniz**
- [node] add edge-node-compat-plugin
- [node] implement edge-handler

### CODE REVIEW

File: packages/node/src/edge-functions/edge-handler-template.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 /* global addEventListener */
 
 function buildUrl(requestDetails) {
-  let proto = requestDetails.headers['x-forwarded-proto'].split(/\b/).shift(); // handling multi-protocol like https,http://...
+  let proto = requestDetails.headers['x-forwarded-proto']?.split(/\b/).shift(); // handling multi-protocol like https,http://...
```
Comment (EndangeredMassa) [inline packages/node/src/edge-functions/edge-handler-template.js]: **suggestion (non-blocking):** revert this change

Just in case this happens, I think we should throw a more explicit error message. I opened a PR with some changes around error handling that covers this case: https://github.com/vercel/vercel/pull/9697

**Review by EndangeredMassa — DISMISSED** [review]

File: packages/node/src/edge-functions/edge-handler-template.js (hunk 1 above)
Comment (cb1kenobi) [reply packages/node/src/edge-functions/edge-handler-template.js]: I not sure we should revert. Now that I think about it, the current code seems bugged as it's possible for the current code to resolve `undefined` which would cause the result to be `'undefined://hostname/path'`, which is certainly not what we want.

Instead of throwing an error, we could just default `proto` to the same protocol as parsed from `requestDetails.url`. That's basically what happens if `x-forwarded-proto` isn't specified.

Comment (Schniz) [reply packages/node/src/edge-functions/edge-handler-template.js]: I added this because the test I added failed for this reason and it seemed better to change this instead of failing for a missing header. We can revert this change tomorrow and I’ll add the header in my test.

Comment (EndangeredMassa) [reply packages/node/src/edge-functions/edge-handler-template.js]: The code as-is on main will at least fail close to the root of the problem (missing `x-forwarded-proto` header). I think it's worth reverting for this PR.

Let's get a better fix in https://github.com/vercel/vercel/pull/9697

Comment (EndangeredMassa) [reply packages/node/src/edge-functions/edge-handler-template.js]: We ended up implementing fallbacks for these values: https://github.com/vercel/vercel/pull/9697

**Review by cb1kenobi — COMMENTED** [review]
Reviewed and tested. Looks good once the merge conflicts are resolved.

**Review by javivelasco — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

**Review by EndangeredMassa — APPROVED** [review]
